test/unit/auth/user-record.spec.ts

```diff
@@ -204,11 +239,270 @@ function getUserInfoWithPhoneNumberJSON(): object {
+    it('should set missing optional fields to null', () => {
+      expect(phoneMultiFactorInfoMissingFields.uid).to.equal(serverResponse.mfaEnrollmentId);
+      expect(phoneMultiFactorInfoMissingFields.displayName).to.be.null;
+      expect(phoneMultiFactorInfoMissingFields.phoneNumber).to.equal(serverResponse.phoneInfo);
+      expect(phoneMultiFactorInfoMissingFields.enrollmentTime).to.be.null;
+      expect(phoneMultiFactorInfoMissingFields.factorId).to.equal('phone');
+    });
+
+    it('should return expected factorId', () => {
+      expect(phoneMultiFactorInfo.factorId).to.equal('phone');
+    });
+
+    it('should throw when modifying readonly factorId property', () => {
+      expect(() => {
+        (phoneMultiFactorInfo as any).factorId = 'other';
+      }).to.throw(Error);
+    });
+
+    it('should return expected displayName', () => {
+      expect(phoneMultiFactorInfo.displayName).to.equal(serverResponse.displayName);
+    });
+
+    it('should throw when modifying readonly displayName property', () => {
+      expect(() => {
+        (phoneMultiFactorInfo as any).displayName = 'Modified';
+      }).to.throw(Error);
+    });
+
+    it('should return expected phoneNumber', () => {
+      expect(phoneMultiFactorInfo.phoneNumber).to.equal(serverResponse.phoneInfo);
+    });
+
+    it('should throw when modifying readonly phoneNumber property', () => {
+      expect(() => {
+        (phoneMultiFactorInfo as any).phoneNumber = '+16505551111';
+      }).to.throw(Error);
+    });
+
+    it('should return expected uid', () => {
+      expect(phoneMultiFactorInfo.uid).to.equal(serverResponse.mfaEnrollmentId);
+    });
+
+    it('should throw when modifying readonly uid property', () => {
+      expect(() => {
+        (phoneMultiFactorInfo as any).uid = 'modifiedEnrollmentId';
+      }).to.throw(Error);
+    });
+
+    it('should return expected enrollmentTime', () => {
+      expect(phoneMultiFactorInfo.enrollmentTime).to.equal(now.toUTCString());
+    });
+
+    it('should throw when modifying readonly uid property', () => {
+      expect(() => {
+        (phoneMultiFactorInfo as any).enrollmentTime = new Date().toISOString();
+      }).to.throw(Error);
+    });
+  });
+
+  describe('toJSON', () => {
+    it('should return expected JSON object', () => {
+      expect(phoneMultiFactorInfo.toJSON()).to.deep.equal({
+        uid: 'enrollmentId1',
+        displayName: 'displayName1',
+        enrollmentTime: now.toUTCString(),
+        phoneNumber: '+16505551234',
+        factorId: 'phone',
+      });
+    });
+
+    it('should return expected JSON object with missing fields set to null', () => {
+      expect(phoneMultiFactorInfoMissingFields.toJSON()).to.deep.equal({
+        uid: 'enrollmentId1',
+        displayName: null,
+        enrollmentTime: null,
+        phoneNumber: '+16505551234',
+        factorId: 'phone',
+      });
+    });
+  });
+});
+
+describe('MultiFactorInfo', () => {
+  const serverResponse: any = {
```

Comment:
Can we use the specific type instead of any?

Reply:
Done.